clusterloader2/pkg/test/test.go

```diff
@@ -31,10 +34,75 @@ func RunTest(f *framework.Framework, conf *api.Config) error {
 	if conf == nil {
 		return fmt.Errorf("test config must be provided")
 	}
-	err := f.CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
-	defer f.DeleteAutomanagedNamespaces()
+
+	ctx := CreateContext(f, state.NewNamespacesState())
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	defer cleanupResources(ctx)
 	if err != nil {
 		return fmt.Errorf("automanaged namespaces creation failed: %v", err)
 	}
+
+	for i := range conf.Steps {
+		executeStep(ctx.Step(i), &conf.Steps[i])
+	}
 	return nil
 }
+
+func executeStep(ctx StepContext, step *api.Step) {
```

Comment:
Ideally, I'd make `Execute()` a method on the `Step` interface, because:

- this would make it possible to have different implementations of running a step (for e.g if someone wants to do something differently like "run only N phases of the step in parallel at a time")
- this would help you eliminate need for  ad-hoc methods like `GetExecutionStep()` that you've currently added to interface just for your implementation